lib/pka_dev.c

```diff
@@ -322,19 +323,17 @@ static int pka_dev_init_ring(pka_dev_ring_t *ring, uint32_t ring_id,
     if (window_ram_split == PKA_SHIM_WINDOW_RAM_SPLIT_ENABLED)
     {
         ring_mem_off  = shim_ring_id * PKA_RING_MEM_1_SPACING;
-        ring_mem_base = ring_mem_off + PKA_RING_MEM_1_BASE;
-        ring_mem_size = PKA_RING_MEM_1_SIZE;
+        ring_mem_base = ring_mem_off + shim->mem_res.alt_wndw_ram_0_base;
     }
     else
     {
         ring_mem_off  = shim_ring_id * PKA_RING_MEM_0_SPACING;
-        ring_mem_base = ring_mem_off + PKA_RING_MEM_0_BASE;
-        ring_mem_size = PKA_RING_MEM_0_SIZE;
+        ring_mem_base = ring_mem_off + shim->mem_res.wndw_ram_base;
     }
 
     ring_window_ram_ptr         = &ring->resources.window_ram;
     ring_window_ram_ptr->base   = ring_mem_base;
-    ring_window_ram_ptr->size   = ring_mem_size;
+    ring_window_ram_ptr->size   = PKA_RING_MEM_SIZE;

```

Comment:
Any reason why we use a macro instead of a value we previously derive from a ACPI or any other calculation based on the window ram mode for example?

Reply:
Earlier we were using `PKA_RING_MEM_0_SIZE` and `PKA_RING_MEM_1_SIZE` which were both defined to be the same value of 16KB. With ACPI we have window ram size as 64 KB and alternate window ram size as 16 KB. Using alternate window ram size (16KB) solves our problem but it doesn't depict a correct design. Hence I removed the above two macros and combined them into a single macro and used it here in the common area as the value in both cases is the same.